src/api-service/__app__/onefuzzlib/workers/pools.py

```diff
@@ -239,6 +230,17 @@ def halt(self) -> None:
 
         self.save()
 
+    def set_state(self, state: PoolState) -> None:
+        if self.state == state:
+            return
+
+        # scalesets should never leave the `halt` state
```

Comment:
This feels a little off, in that I expect `set_state()` to only be responsible for (1) avoiding spurious "updates" of cycle transitions, and (2) managing saving remote state.

Here, we're also dropping requests for illegal state transitions. It'd be nice if we had a higher-level method that was responsible for such enforcement, then called this method to execute the actual state change.

Reply:
Per offline discussion, we'll revisit this concern holistically by investigating a more structured use of state machines.